packages/grafana-ui/src/components/uPlot/context.ts

```diff
@@ -127,27 +126,25 @@ export const usePlotData = (): PlotDataAPI => {
 };
 
 export const buildPlotContext = (
-  isPlotReady: boolean,
   canvasRef: any,
   data: AlignedFrameWithGapTest,
   registerPlugin: any,
-  getPlotInstance: () => uPlot
+  getPlotInstance: () => uPlot | undefined
 ): PlotContextType => {
   return {
-    isPlotReady,
     canvasRef,
     data,
     registerPlugin,
     getPlotInstance,
-    getSeries: () => getPlotInstance().series,
+    getSeries: () => getPlotInstance()!.series,
     getCanvas: () => ({
-      width: getPlotInstance().width,
-      height: getPlotInstance().height,
+      width: getPlotInstance()!.width,
+      height: getPlotInstance()!.height,
       plot: {
-        width: getPlotInstance().bbox.width / window.devicePixelRatio,
-        height: getPlotInstance().bbox.height / window.devicePixelRatio,
-        top: getPlotInstance().bbox.top / window.devicePixelRatio,
-        left: getPlotInstance().bbox.left / window.devicePixelRatio,
+        width: getPlotInstance()!.bbox.width / window.devicePixelRatio,
+        height: getPlotInstance()!.bbox.height / window.devicePixelRatio,
+        top: getPlotInstance()!.bbox.top / window.devicePixelRatio,
+        left: getPlotInstance()!.bbox.left / window.devicePixelRatio,
```

Comment:
could be a bit more terse by aliasing above the return:

```js
const plot = getPlotInstance()!,
      bbox = plot.bbox,
      pxRatio = window.devicePixelRatio;
```

Reply:
yea, nice to get rid of all the calls to getPlotInstance()!   :)

Reply:
👍